ports/stm32/mboot/README.md

```diff
@@ -209,6 +219,33 @@ and signed firmware, and can be deployed via USB DFU, or by copying it to the de
 internal filesystem (if `MBOOT_FSLOAD` is enabled). `firmware.dfu` is still unencrypted
 and can be directly flashed with jtag etc.
 
+Retrieving the mboot version in micropython
+--------------------------------------
+The following python code retrieves the version of mboot currently on the device:
+```python
+import stm
+
+def get_mboot_version(include_opts=True):
+    MBOOT_BASE = const(0x0800_0000)
+    MBOOT_LEN = const(32 * 1024)  # this will need to updated if mboot is a different size
+    VER_LEN = const(64)  # this will need to be updated if the length/location of version string changes
+    base = const(MBOOT_BASE + MBOOT_LEN - VER_LEN)
+    s = ''
+    for i in range(VER_LEN):
+        c = stm.mem8[base + i]
+        if c == 0xFF or c == 0:  # have hit empty flash or a null-terminator
```

Comment:
I don't think checking for null is needed.  The way the string is generated, null won't ever be there.